code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	..() //needed so the rune still gets stored by the crossbreed
+	qdel(holotile)
+
+
+
+
+
+/obj/item/slimecross/warping/pyrite
+	colour = "pyrite"
+	runepath = /obj/effect/warped_rune/pyritespace
+	effect_desc = "draws a rune that will randomly color whoever steps on it"
+
+/obj/effect/warped_rune/pyritespace
+	desc = "Who shall we be today? they asked, but not even the canvas would answer."
+	icon_state = "colorune"
+
+/obj/effect/warped_rune/pyritespace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/colored)
+
+///colors whoever steps on the rune randomly
+/obj/effect/warped_rune/pyritespace/proc/colored()
+	for(var/mob/living/L in T)
+		L.color = rgb(rand(0,255),rand(0,255),rand(0,255))
+
+
+/* Will make anyone on the rune do much MUCH more punch damage. Doesn't boost normal melee weapon damage*/
+
+/obj/item/slimecross/warping/red
+	colour = "red"
+	runepath = /obj/effect/warped_rune/redspace
+	effect_desc = "draw a rune severely increasing your punch damage. Only works on one person at a time."
+    drawing_time = 100 //longer to draw so you don't draw it in the middle of a fight.
+
+/obj/effect/warped_rune/redspace
+	desc = "Progress is made through adversity, power is obtained through violence"
+	icon_state = "rage_rune"
+	runepath = /obj/effect/warped_rune/redspace
+	var/mob/living/carbon/human/H
+	var/datum/species/og_punch_low //used to remember the original low punch force of H
+	var/og_punch_high //used to remember the original high punch force of H
+	var/rage = FALSE //here to check if someone is already enraged on the rune
+	var/og_threshold
+
+/obj/effect/warped_rune/redspace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/angry_in)
+	RegisterSignal(T,COMSIG_ATOM_EXITED,.proc/angry_out)
+
+
+/obj/effect/warped_rune/redspace/proc/angry_in()
+
+	if(rage) //don't buff anyone if someone is already buffed by the rune.
+		return
+
+	for(var/mob/living/carbon/human/A in T)
+		og_punch_low = A.dna.species.punchdamagelow
+		og_punch_high = A.dna.species.punchdamagehigh
+		og_threshold = A.dna.species.punchstunthreshold
+		H = A
+		RegisterSignal(H,COMSIG_HUMAN_MELEE_UNARMED_ATTACK,.proc/blind_rage)//triggers whenever H attacks
+		H.dna.species.punchdamagelow = 40
+		H.dna.species.punchdamagehigh = 50
+		H.dna.species.punchstunthreshold = 50 //so each punch doesn't insta stun
+		rage = TRUE
+
+
+/obj/effect/warped_rune/redspace/proc/angry_out()
+	var/turf/HT = get_turf(H)
+	if(HT == T && locate(runepath) in T)
+		return
+	H.dna.species.punchdamagelow = og_punch_low
+	H.dna.species.punchdamagehigh = og_punch_high
+	H.dna.species.punchstunthreshold = og_threshold 
+	UnregisterSignal(H,COMSIG_HUMAN_MELEE_UNARMED_ATTACK)
+	rage = FALSE //allows another person to enter rage mode on the rune.
+
+///triggers whenever the enraged player punches something. lowers their punch damage and deals stamina damage to themselves.
+/obj/effect/warped_rune/redspace/proc/blind_rage()
+	H.dna.species.punchdamagelow -= 10
```

Comment:
hm yes today I will make my punchdamagelow be a negative number

honestly, this stuff isn't really needed, seeing as how you can only really punch people who are stupid enough to get close to the rune while you're on it and you lose your powers if you step off of the rune. you can also just get around this downside (well, apart from the stamina damage) by stepping off of and then back onto the rune. also, for a proc named blind_rage(), I was kind of expecting a blindness or a blurriness debuff.